src/entities/join/index.ts

```diff
@@ -9,28 +10,16 @@ export enum JoinKind {
     Proportional = 'Proportional',
 }
 
-// Returned from API and used as input
-export type PoolState = {
-    id: Hex;
-    address: Address;
-    type: string;
-    tokens: {
-        address: Address;
-        decimals: number;
-        index: number;
-    }[];
-};
-
 // This will be extended for each pools specific input requirements
 export type BaseJoinInput = {
     chainId: number;
-    rpcUrl?: string;
+    rpcUrl: string;
```

Comment:
I made `rpcUrl` optional because viem provides a default value in case the user wants to use it.
I'm not against making it required, but I'd say that it doesn't have to be.

Reply:
Good point. I've definitely had issues where default RPCs fail and its confusing so making it explicit to provide one might help to make that clearer (but maybe thats because original SDK is doing intensive multicalls). Could make it required for now and fine to change if we feel we need to.